Source/Charts/Renderers/LineChartRenderer.swift

```diff
@@ -751,4 +767,113 @@ open class LineChartRenderer: LineRadarRenderer
         
         context.restoreGState()
     }
+
+    /// Generates the path that is used for gradient drawing.
+    private func generateGradientLinePath(dataSet: LineChartDataSetProtocol, fillMin: CGFloat, from: Int, to: Int, matrix: CGAffineTransform) -> CGPath
+    {
+        let phaseX = CGFloat(animator.phaseX)
+        let phaseY = CGFloat(animator.phaseY)
+
+        var e: ChartDataEntry!
+
+        let generatedPath = CGMutablePath()
+        e = dataSet.entryForIndex(from)
+        if e != nil
+        {
+            generatedPath.move(to: CGPoint(x: CGFloat(e.x), y: CGFloat(e.y) * phaseY), transform: matrix)
+        }
+
+        // create a new path
+        let to = Int(ceil(CGFloat(to - from) * phaseX + CGFloat(from)))
+        for i in (from + 1)..<to
```

Comment:
`_xBounds.max` and `_xBounds.min` can be zeros this for is crashing the app in this case with `Fatal error: Can't form Range with upperBound < lowerBoun`

Reply:
It needs `if dataSet.entryCount > 1` somewhere.